backend/endpoints/ratings.py

```diff
@@ -0,0 +1,114 @@
+import os
+from fastapi import APIRouter, Depends, HTTPException
+from sqlalchemy.orm import Session
+from sqlalchemy.sql import func
+from src.database.database import get_db
+from src.database.models import Video, VideoRating, User
+from src.database.schemas import RatingCreate, Rating, VideoWithRatings
+from src.user.auth import get_current_user
+import random
+from typing import Optional
+from pydantic import ValidationError
+
+
+router = APIRouter()
+
+@router.get("/random", response_model=VideoWithRatings)
+async def get_random_video(
+    db: Session = Depends(get_db),
+    current_user: User = Depends(get_current_user)
+):
+    # Get all video IDs that the current user hasn't rated yet
+    rated_video_ids = db.query(VideoRating.video_id).filter(VideoRating.user_id == current_user.id)
+    unrated_videos = db.query(Video).filter(~Video.id.in_(rated_video_ids)).all()
+    
+    if not unrated_videos:
+        raise HTTPException(status_code=404, detail="No unrated videos found")
+    
+    # Select a random video from the unrated videos
+    random_video = random.choice(unrated_videos)
+    
+    # Get the average rating and rating count for this video
+    rating_stats = db.query(
+        func.avg(VideoRating.rating).label('average_rating'),
+        func.count(VideoRating.id).label('ratings_count')
+    ).filter(VideoRating.video_id == random_video.id).first()
+
+    # Convert the local file path to a web-accessible URL
+    video_filename = os.path.basename(random_video.url)
+    video_url = f"/app/videos/{video_filename}"
```

Comment:
Should be a YouTube link, just some gangam style or something until we figure out the youtube upload mechanism. That way we can ensure that the player supports playing a video from Youtube.
Variable `video_filename` gets the url which currently is a local path, but later will be the actual youtube path.